mobile/src/main/java/ch/epfl/sdp/mobile/ui/setting/EditProfileImageDialog.kt

```diff
@@ -0,0 +1,127 @@
+   the dialog when it's filled for the first time. On the following recompositions, this new size
+   is applied as the constraints to the root of the hierarchy and some elements might not be able
+   to occupy some space they need.
+   Applying Modifier.fillMaxSize() makes sure we "reserve" this space and that the window will
+   never force us to shrink our content.
+  */
+  Box(modifier.fillMaxSize(), Alignment.Center) {
+    val strings = LocalLocalizedStrings.current
+
+    Dialog(
+        cancelContent = { Text(text = strings.settingEditCancel) },
+        confirmContent = { Text(text = strings.settingEditSave) },
+        onCancelClick = state::onCancelClick,
+        onConfirmClick = state::onSaveClick) {
+      Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
+        EditSettingPicture(backgroundColor = state.backgroundColor, emoji = state.emoji)
+        Text(strings.settingProfileNameLabel, style = MaterialTheme.typography.subtitle1)
+        LazyRow() { items(items = emojis) { item -> SelectEmojiItem(state, item) } }
+        LazyRow() {
+          items(items = Color.asList()) { item -> SelectBackgroundColorItem(state, item) }
+        }
+      }
+    }
+  }
+}
+/**
+ * Composes the edit settings picture which shows the edit changes
+ * @param backgroundColor background color for setting image
+ * @param emoji the emoji string
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun EditSettingPicture(backgroundColor: Color, emoji: String, modifier: Modifier = Modifier) {
+
+  Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = Modifier.fillMaxWidth()) {
+    Box(
+        modifier = modifier.size(118.dp).background(backgroundColor.toColor(), CircleShape),
+        contentAlignment = Alignment.Center,
+    ) { Text(emoji, style = MaterialTheme.typography.h3) }
+  }
+}
+
+/**
+ * Select Emoji Item
+ * @param state which is [EditProfileImageDialogState] and modifies it
+ * @param emoji String which to be changed
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun SelectEmojiItem(
+    state: EditProfileImageDialogState,
+    emoji: String,
+    modifier: Modifier = Modifier
+) {
+  Box(
+      modifier =
+          if (state.emoji == emoji)
+              modifier.size(72.dp).border(BorderStroke(4.dp, Green800), CircleShape)
+          else modifier.size(72.dp).clickable { state.emoji = emoji },
+      contentAlignment = Alignment.Center,
+  ) { Text(emoji, style = MaterialTheme.typography.h4) }
+}
+
+/**
+ * Select Background color of profile Image
+ * @param state which is [EditProfileImageDialogState] and modifies it
+ * @param backgroundColor the current color in item
+ * @param modifier the [Modifier] for this composable.
+ */
+@Composable
+fun SelectBackgroundColorItem(
+    state: EditProfileImageDialogState,
+    backgroundColor: Color,
+    modifier: Modifier = Modifier
+) {
```

Comment:
(same as above) This function should be private. Moreover, I feel like it receives a state that's too big. What would you think of making it have the following signature ?

```kotlin
private fun SelectBackgroundColorItem(
  color: Color,
  selected: Boolean,
  onClick: () -> Unit,
  modifier: Modifier = Modifier,
)
```

Reply:
is done